src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```
@@ -0,0 +1,279 @@
+
+        } catch (JsonProcessingException e) {
+            log.error(SERIALIZATION_ERROR_MSG, e);
+            return  createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        } catch (Exception e ) {
+            log.error(GENERIC_ERROR_MSG + transactionId);
+            return createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+
+        log.info("END - XPay Request Payment Authorization ");
+        return createxPayAuthResponse(null, HttpStatus.OK, paymentRequestEntity.getGuid());
+    }
+
+    @Async
+    private void executeXPayAuthorizationCall(AuthPaymentXPayRequest xPayRequest, PaymentRequestEntity paymentRequestEntity, String transactionId) throws Exception {
+        log.info("START - execute XPay payment authorization call for transactionId: " + transactionId);
+        AuthPaymentXPayResponse response;
+        try {
+            response = service.postForObject(xPayRequest);
+            if(Objects.nonNull(response)) {
+                paymentRequestEntity.setTimeStamp(String.valueOf(response.getTimeStamp()));
+                if(Objects.isNull(response.getErrore())) {
+                    paymentRequestEntity.setXpayHtml(response.getHtml());
+                    paymentRequestEntity.setAuthorizationOutcome(true);
+                } else {
+                    paymentRequestEntity.setErrorCode(String.valueOf(response.getErrore().getCodice()));
+                    paymentRequestEntity.setErrorMessage(response.getErrore().getMessaggio());
+                    paymentRequestEntity.setStatus(DENIED.name());
+                }
+            } else {
+                log.warn("Response received from xPay is null");
+                throw new Exception();
+            }
+        } catch (ResourceAccessException e) {
+            log.warn("Timeout exception for xpay api request call");
+            throw e;
```

There is no need to re-throw these exceptions since the caller has a generic catch() clause for Exception type.
You can simply write the exception stack-trace into the log.error clause (another thing: it is better to log exceptions at log.error level)